packages/react-ui-core/src/Highlighter/Highlighter.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import { randomId } from '@rentpath/react-ui-utils'
+
+@themed(/^Highlighter/)
+
+export default class Highlighter extends Component {
+  static propTypes = {
+    pattern: PropTypes.string,
+    children: PropTypes.string,
+    theme: PropTypes.object,
+    indexHighlighted: PropTypes.number,
+  }
+
+  static defaultProps = {
+    text: '',
+    children: '',
+    theme: {},
+  }
+
+  get indexHighlighted() {
+    return this.props.indexHighlighted + 1
+  }
+
+  renderHighlightedText() {
+    const { pattern, children } = this.props
+
+    const segments = children.split(pattern)
+
+    return segments.reduce((prev, curr, index) => [prev,
+      this.renderHighlightedSegment(index),
+      curr])
+  }
+
+  renderHighlightedSegment(index) {
+    const { pattern, theme, indexHighlighted } = this.props
+
+    if (indexHighlighted === undefined || index === indexHighlighted) {
```

Comment:
What happens if `indexHighlighted` is null?

Reply:
The propTypes enforce it as a number, so as far as I understand, it could only be `undefined` or a number at this point.

Reply:
If that's the case, can you add a test for `null`?